diskmaker/discovery/discovery.go

```diff
@@ -197,7 +198,26 @@ func getDiscoverdDevices(blockDevices []internal.BlockDevice) []v1alpha1.Discove
 		discoveredDevices = append(discoveredDevices, discoveredDevice)
 	}
 
-	return discoveredDevices
+	return uniqueDevices(discoveredDevices)
+}
+
+// uniqueDevices removes duplicate devices from the list using DeviceID as a key
+// TODO: remove this and use lsblk with -M flag once base images are updated and lsblk v2.34 or higher is available
+// See: https://github.com/util-linux/util-linux/blob/3be31a106c52e093928afbea2cddbdbe44cfb357/Documentation/releases/v2.34-ReleaseNotes#L18
+func uniqueDevices(sample []v1alpha1.DiscoveredDevice) []v1alpha1.DiscoveredDevice {
+	var unique []v1alpha1.DiscoveredDevice
+	type key struct{ value string }
+	m := make(map[key]int)
+	for _, v := range sample {
+		k := key{v.DeviceID}
+		if i, ok := m[k]; ok {
+			unique[i] = v
```

Comment:
nit: why did we make a key and type for it? Couldn't we have just used `DeviceId` via something like `map[string]int` ?

Reply:
But another thing is - `deviceID` may not be available always. Could we not have used `v.path` for uniqueness?

Reply:
>nit: why did we make a key and type for it? Couldn't we have just used DeviceId via something like map[string]int ?

Sure, but if we would need more criteria for uniqueness instead of just one string using custom type would allow that, like this:

```
...
	type key struct{ value string; value2 string }
	m := make(map[key]int)
	for _, v := range sample {
		k := key{v.DeviceID, v.Path}
...
```

>But another thing is - deviceID may not be available always. Could we not have used v.path for uniqueness?

Correct, `deviceID` can be set to `""` - I missed that. That's even worse with this unique filter because that would hide all devices (except one) that have `deviceID` set to `""`. I think it makes sense to use what I described above and let `uniqueDevices()` use both `v.Path` and `v.DeviceID`?

Reply:
>nit: why did we make a key and type for it? Couldn't we have just used DeviceId via something like map[string]int ?

Sure, but if we would need more criteria for uniqueness instead of just one string using custom type would allow that, like this:

```
...
	type key struct{ value string; value2 string }
	m := make(map[key]int)
	for _, v := range sample {
		k := key{v.DeviceID, v.Path}
...
```

>But another thing is - deviceID may not be available always. Could we not have used v.path for uniqueness?

Correct, `deviceID` can be set to `""` - I missed that. That's even worse with this unique filter because that would hide all devices (except one) that have `deviceID` set to `""`. I think it makes sense to use what I described above and let `uniqueDevices()` use both `v.Path` and `v.DeviceID`?

Reply:
should work.